govcd/lbappprofile.go

```diff
@@ -11,59 +11,59 @@ import (
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
 )
 
-// CreateLBAppProfile creates a load balancer application profile based on mandatory fields. It is a
+// CreateLbAppProfile creates a load balancer application profile based on mandatory fields. It is a
 // synchronous operation. It returns created object with all fields (including ID) populated or an error.
-func (eGW *EdgeGateway) CreateLBAppProfile(lbAppProfileConfig *types.LBAppProfile) (*types.LBAppProfile, error) {
-	if err := validateCreateLBAppProfile(lbAppProfileConfig); err != nil {
+func (egw *EdgeGateway) CreateLbAppProfile(lbAppProfileConfig *types.LbAppProfile) (*types.LbAppProfile, error) {
+	if err := validateCreateLbAppProfile(lbAppProfileConfig, egw); err != nil {
 		return nil, err
 	}
 
-	httpPath, err := eGW.buildProxiedEdgeEndpointURL(types.LBAppProfilePath)
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.LbAppProfilePath)
 	if err != nil {
 		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
 	}
 	// We expect to get http.StatusCreated or if not an error of type types.NSXError
-	resp, err := eGW.client.ExecuteRequestWithCustomError(httpPath, http.MethodPost, types.AnyXMLMime,
+	resp, err := egw.client.ExecuteRequestWithCustomError(httpPath, http.MethodPost, types.AnyXMLMime,
 		"error creating load balancer application profile: %s", lbAppProfileConfig, &types.NSXError{})
 	if err != nil {
 		return nil, err
 	}
 
 	// Location header should look similar to:
 	// [/network/edges/edge-3/loadbalancer/config/applicationprofiles/applicationProfile-4]
-	lbAppProfileID, err := extractNSXObjectIDFromPath(resp.Header.Get("Location"))
+	lbAppProfileID, err := extractNsxObjectIdFromPath(resp.Header.Get("Location"))
 	if err != nil {
 		return nil, err
 	}
 
-	readAppProfile, err := eGW.ReadLBAppProfileByID(lbAppProfileID)
+	readAppProfile, err := egw.GetLbAppProfileById(lbAppProfileID)
 	if err != nil {
 		return nil, fmt.Errorf("unable to retrieve application profile with ID (%s) after creation: %s",
 			readAppProfile.ID, err)
```

Comment:
readAppProfile.ID seems to be accessing a nil pointer

Reply:
Yes. Changed it.